src/newrelic/processor.go

```diff
@@ -456,6 +462,49 @@ func (p *Processor) processConnectAttempt(rep ConnectAttempt) {
 		NewHarvest(time.Now(), app.connectReply.EventHarvestConfig.EventConfigs), p.processorHarvestChan)
 }
 
+func processLogEventLimits(app *App) {
+
+	if nil == app {
+		log.Warnf("processLogEventLimits() called with *App == nil")
+		return
+	}
+
+	if nil == app.info {
+		log.Warnf("processLogEventLimits() called with app.info == nil")
+		return
+	}
+
+	if nil == app.connectReply {
+		log.Warnf("processLogEventLimits() called with  app.connectReply == nil")
+		return
+	}
+
+	// need to compare agent limits to limits returned from the collector
+	// and choose the smallest value
+	agentLogLimit := app.info.AgentEventLimits.LogEventConfig.Limit
+	agentReportPeriod := float64(limits.DefaultReportPeriod)
+
+	collectorLogLimit := app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit
+	collectorReportPeriod := float64(app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.ReportPeriod)
+
+	// convert agent log limit to sampling period used for log events
+	agentLogLimit = int((float64(agentLogLimit) * collectorReportPeriod) / agentReportPeriod)
+
+	log.Debugf("handling log limits: agent_report_period = %f collector_report_period = %f", agentReportPeriod, collectorReportPeriod)
+	log.Debugf("handling log limits: agent_log_limit = %d agentLogLimit = %d", agentLogLimit, collectorLogLimit)
+
+	finalLogLimit := collectorLogLimit
+	if agentLogLimit < collectorLogLimit {
+		finalLogLimit = agentLogLimit
+	}
+
+	// store final log limit in reply object which is used by rest of code to
+	// establish harvest limits
+	app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit = finalLogLimit
```

Comment:
Just verifying, this number is never sent to the collector, right?  It's just what is used internally?

Reply:
Yes the connect reply is never used to create a connect request.  When the connect request is create in `app.go::ConnectPayloadInternal() `it refers to the `AgentEventLimits` sent in from the agent.  These limits are not being modified.
The limit being modified is used when creating the pool for events in `processing.go::harvestByType()` when a harvest cycle completes and in `harvest.go::NewHarvest()` when the harvest is being initialized the first time.